src/cpp/session/modules/SessionRmdNotebook.R

```diff
@@ -38,6 +38,12 @@ assign(".rs.notebookVersion", envir = .rs.toolsEnv(), "1.0")
    "^[\t >]*```+\\s*$"
 })
 
+.rs.addFunction("reYamlOptChunkBegin", function()
+{
+   "^#\\| [a-zA-Z_-]+:\\s*.+$"
```

Comment:
One thing we'll have to be careful of here is that the inline chunk options syntax is _very_ liberal. From https://yihui.org/knitr/options/#chunk-options, it looks like you could even just include the chunk header inline:

`````
```{r}
#| my-chunk, echo = FALSE, fig.width = 10,
#| fig.cap = "This is a long long
#|   long long caption."

plot(cars)
```
`````

Hopefully this will not be common!

Reply:
More concretely, I think we need to adjust the pattern here to tolerate:

- Option names containing a `.`, e.g. `fig.width`
- Comments placed after the option value.

An example test case:

`````
---
title: "Untitled"
output: html_document
date: "2022-04-28"
---

```{r}
#| eval: true     # this is a comment
#| fig.width: 42  # this is also a comment
knitr::opts_current$get("fig.width")
```
`````

Reply:
Oh, I was reviewing those docs but I don't think I even saw that you could have multiple options per line, I'll have to double check as I may not have accounted for that